### ISSUE TITLE
test failure on macOS 10.13

### ISSUE THREAD

**Issue by chdiza** [vim/vim#3953]
Some commit from today causes a test failure on macOS 10.13:
```
VIMRUNTIME=../../runtime; export VIMRUNTIME;  ../vim -f  -u unix.vim -U NONE --noplugin --not-a-term -S runtest.vim test_alot_latin.vim --cmd 'au SwapExists * let v:swapchoice = "e"' > /dev/null

/bin/sh: line 1: 63462 Segmentation fault: 11  ../vim -f -u unix.vim -U NONE --noplugin --not-a-term -S runtest.vim test_alot_latin.vim --cmd 'au SwapExists * let v:swapchoice = "e"' > /dev/null
                                  make[2]: *** [test_alot_latin.res] Error 139
                                                                              make[1]: *** [scripttests] Error 2
                                make: *** [test] Error 2
```
And afterwards my terminal is messed up.

**Comment by dpelle** [comment]
It also fails in CI.

On my Linux machine with valgrind, I see this using vim-8.1.912 on Linux when running `make test_alot`:
```
==7623== Memcheck, a memory error detector
==7623== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==7623== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==7623== Command: ../vim -f -u unix.vim -U NONE --noplugin --not-a-term -S runtest.vim test_alot_latin.vim --cmd au\ SwapExists\ *\ let\ v:swapchoice\ =\ "e"
==7623== Parent PID: 7622
==7623== 
==7623== Warning: set address range perms: large range [0x16e97040, 0x279f17a0) (undefined)
==7623== Warning: set address range perms: large range [0x37d49028, 0x488a37b8) (noaccess)
==7623== Warning: set address range perms: large range [0x72f58040, 0x8c06c670) (undefined)
==7623== Warning: set address range perms: large range [0x16e97028, 0x279f17b8) (noaccess)
==7623== Warning: set address range perms: large range [0x59e43028, 0x72f57688) (noaccess)
==7623== Warning: set address range perms: large range [0x8c06d040, 0xb1a182a0) (undefined)
==7623== Warning: set address range perms: large range [0x72f58028, 0x8c06c688) (noaccess)
==7623== Warning: set address range perms: large range [0xd73c42a0, 0xea0a66f0) (undefined)
==7623== Warning: set address range perms: large range [0x279f2028, 0x4d39d2b8) (noaccess)
==7623== Warning: set address range perms: large range [0xea0a7040, 0x1227346f0) (undefined)
==7623== Warning: set address range perms: large range [0x8c06d028, 0xb1a182b8) (noaccess)
==7623== Warning: set address range perms: large range [0x15adc26f0, 0x177115b60) (undefined)
==7623== Warning: set address range perms: large range [0xb1a19028, 0xea0a6708) (noaccess)
==7623== Warning: set address range perms: large range [0x59e43040, 0xae823b60) (undefined)
==7623== Warning: set address range perms: large range [0xea0a7028, 0x122734708) (noaccess)
==7623== Warning: set address range perms: large range [0x1cbaf6b60, 0x1f5ff3c10) (undefined)
==7623== Warning: set address range perms: large range [0x122735028, 0x177115b78) (noaccess)
==7623== Warning: set address range perms: large range [0x1f5ff4040, 0x274ed1c10) (undefined)
==7623== Warning: set address range perms: large range [0x59e43028, 0xae823b78) (noaccess)
==7623== Warning: set address range perms: large range [0x12d701c10, 0x16ce7d310) (undefined)
==7623== Warning: set address range perms: large range [0x177116028, 0x1f5ff3c28) (noaccess)
==7623== Warning: set address range perms: large range [0x274ed2040, 0x33352b310) (undefined)
==7623== Warning: set address range perms: large range [0x1f5ff4028, 0x274ed1c28) (noaccess)
==7623== Warning: set address range perms: large range [0x3f1b85310, 0x450ebe590) (undefined)
==7623== Warning: set address range perms: large range [0xae824028, 0x16ce7d328) (noaccess)
==7623== Warning: set address range perms: large range [0xbc63040, 0x295f5590) (undefined)
==7623== Invalid write of size 8
==7623==    at 0x4C367E3: memmove (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==7623==    by 0x26C0BF: memmove (string_fortified.h:40)
==7623==    by 0x26C0BF: addstate_here (regexp_nfa.c:4794)
==7623==    by 0x26C6BB: nfa_regmatch (regexp_nfa.c:6794)
==7623==    by 0x26ED2A: nfa_regtry (regexp_nfa.c:7006)
==7623==    by 0x26F1CB: nfa_regexec_both (regexp_nfa.c:7200)
==7623==    by 0x273DB3: vim_regexec_multi (regexp.c:8245)
==7623==    by 0x288CDF: searchit (search.c:751)
==7623==    by 0x17FDC3: search_cmn (evalfunc.c:10280)
==7623==    by 0x18001B: f_search (evalfunc.c:10415)
==7623==    by 0x1845C8: call_internal_func (evalfunc.c:1117)
==7623==    by 0x2E2727: call_func (userfunc.c:1521)
==7623==    by 0x2E37BC: get_func_tv (userfunc.c:455)
==7623==    by 0x2E64BD: ex_call (userfunc.c:3185)
==7623==    by 0x1A9C8B: do_one_cmd (ex_docmd.c:2525)
==7623==    by 0x1A9C8B: do_cmdline (ex_docmd.c:1033)
==7623==    by 0x16961F: assert_fails (eval.c:9503)
==7623==    by 0x17CC48: f_assert_fails (evalfunc.c:1624)
==7623==    by 0x1845C8: call_internal_func (evalfunc.c:1117)
==7623==    by 0x2E2727: call_func (userfunc.c:1521)
==7623==    by 0x2E37BC: get_func_tv (userfunc.c:455)
==7623==    by 0x2E64BD: ex_call (userfunc.c:3185)
==7623==    by 0x1A9C8B: do_one_cmd (ex_docmd.c:2525)
==7623==    by 0x1A9C8B: do_cmdline (ex_docmd.c:1033)
==7623==    by 0x2E2EE5: call_user_func (userfunc.c:968)
==7623==    by 0x2E2EE5: call_func (userfunc.c:1502)
==7623==    by 0x2E37BC: get_func_tv (userfunc.c:455)
==7623==    by 0x2E64BD: ex_call (userfunc.c:3185)
==7623==    by 0x1A9C8B: do_one_cmd (ex_docmd.c:2525)
==7623==    by 0x1A9C8B: do_cmdline (ex_docmd.c:1033)
==7623==  Address 0x295f5590 is 0 bytes after a block of size 496,575,824 alloc'd
==7623==    at 0x4C2FB0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==7623==    by 0x2048D0: lalloc (misc2.c:944)
==7623==    by 0x26C08E: addstate_here (regexp_nfa.c:4790)
==7623==    by 0x26C6BB: nfa_regmatch (regexp_nfa.c:6794)
==7623==    by 0x26ED2A: nfa_regtry (regexp_nfa.c:7006)
==7623==    by 0x26F1CB: nfa_regexec_both (regexp_nfa.c:7200)
==7623==    by 0x273DB3: vim_regexec_multi (regexp.c:8245)
==7623==    by 0x288CDF: searchit (search.c:751)
==7623==    by 0x17FDC3: search_cmn (evalfunc.c:10280)
==7623==    by 0x18001B: f_search (evalfunc.c:10415)
==7623==    by 0x1845C8: call_internal_func (evalfunc.c:1117)
==7623==    by 0x2E2727: call_func (userfunc.c:1521)
==7623==    by 0x2E37BC: get_func_tv (userfunc.c:455)
==7623==    by 0x2E64BD: ex_call (userfunc.c:3185)
==7623==    by 0x1A9C8B: do_one_cmd (ex_docmd.c:2525)
==7623==    by 0x1A9C8B: do_cmdline (ex_docmd.c:1033)
==7623==    by 0x16961F: assert_fails (eval.c:9503)
==7623==    by 0x17CC48: f_assert_fails (evalfunc.c:1624)
==7623==    by 0x1845C8: call_internal_func (evalfunc.c:1117)
==7623==    by 0x2E2727: call_func (userfunc.c:1521)
==7623==    by 0x2E37BC: get_func_tv (userfunc.c:455)
==7623==    by 0x2E64BD: ex_call (userfunc.c:3185)
==7623==    by 0x1A9C8B: do_one_cmd (ex_docmd.c:2525)
==7623==    by 0x1A9C8B: do_cmdline (ex_docmd.c:1033)
==7623==    by 0x2E2EE5: call_user_func (userfunc.c:968)
==7623==    by 0x2E2EE5: call_func (userfunc.c:1502)
==7623==    by 0x2E37BC: get_func_tv (userfunc.c:455)
==7623==    by 0x2E64BD: ex_call (userfunc.c:3185)
==7623== 
==7623== 
==7623== Process terminating with default action of signal 11 (SIGSEGV): dumping core
==7623==    at 0x6F0C187: kill (syscall-template.S:78)
==7623==    by 0x242756: may_core_dump (os_unix.c:3289)
==7623==    by 0x242756: mch_exit (os_unix.c:3298)
==7623==    by 0x3206D7: getout (main.c:1552)
==7623==    by 0x6CC088F: ??? (in /lib/x86_64-linux-gnu/libpthread-2.27.so)
==7623==    by 0x4C367E2: memmove (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
```

**Comment by brammool** [comment]
> It also fails in CI.
> 
> On my Linux machine with valgrind, I see this using vim-8.1.912 on Linux when running `make test_alot`:

This stress testing runs over the CI limits.

I made a change in 8.1.0913 to use 'maxmempattern' also with the new
regexp engine.  Let's see if that fixes this problem.
At least I can't reproduce this valgrind error.


-- 
"I've been teaching myself to play the piano for about 5 years and now write
most of my songs on it, mainly because I can never find any paper."
		Jeff Lynne, ELO's greatest hits

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by ichizok** [comment]
This issue is fixed by v8.1.0913.
